.github/workflows/main.yml

```diff
@@ -0,0 +1,75 @@
+name: Build
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+    branches:
+      - develop
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name || github.run_id }}
+  cancel-in-progress: true
```

Comment:
cancel in progress with a workflow that relies on external resources (a k8s cluster for testing) is usually a bad idea - it can skip the cleanup of resources

Reply:
Indeed, I was looking at ways of cleaning up on cancellation, but that seems to be available only on javascript actions (https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runspost)

Reply:
there is a specific behaviour when a cancel get triggered, see https://docs.github.com/en/actions/managing-workflow-runs/canceling-a-workflow#steps-github-takes-to-cancel-a-workflow-run

Reply:
I removed the concurrency configuration since the checks are quick